internal/cluster/cluster.go

```diff
@@ -1101,22 +1102,23 @@ func (m Manager) PermanentClustersDeletion(ctx context.Context, olderThan strfmt
 		if !deleteFromDB {
 			continue
 		}
-
-		cluster := c // Avoid passing loop variable by ref
-		if reply := db.Delete(&cluster); reply.Error != nil {
-			m.log.WithError(reply.Error).Warnf("Failed deleting cluster from db %s", c.ID.String())
-		} else if reply.RowsAffected > 0 {
-			m.log.Debugf("Deleted %s cluster from db", reply.RowsAffected)
-		}
-
-		if err := common.DeleteRecordsByClusterID(db, *c.ID, []interface{}{
+		modelsToDelete := []interface{}{
 			&models.Event{},
 			&models.MonitoredOperator{},
 			&models.ClusterNetwork{},
 			&models.ServiceNetwork{},
 			&models.MachineNetwork{},
-		}); err != nil {
-			m.log.WithError(err).Warnf("Failed deleting cluster records from db for cluster %s", c.ID.String())
+		}
+		for _, model := range modelsToDelete {
+			if err := common.DeleteRecordsByClusterID(m.db.Unscoped(), *c.ID, []interface{}{model}); err != nil {
```

Comment:
why is unscoped used here? shouldn't our GC handle that? 
We want to keep all the elements in order to investigate them in case of errors

Reply:
This is a function for permanent deletion

Reply:
right, missed that.